### PR TITLE
PPTP-993 : General Partnership GRS Integration

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**

### CODE REVIEW

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/CustomerDetailsSpec.scala [hunk 1]
```diff
@@ -49,7 +49,18 @@ class CustomerDetailsSpec
...
+        customerDetails.customerType mustBe CustomerType.Organisation
 
+        customerDetails.individualDetails mustBe None
+
+        customerDetails.organisationDetails.get.organisationType mustBe Some(
+          pptPartnershipDetails.organisationType.get.toString
+        )
+        customerDetails.organisationDetails.get.organisationName mustBe "TODO"
       }
     }
```
Comment (tmc-ee) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/CustomerDetailsSpec.scala:66]: Can we add a test for exceptions? e.g. `incorporation details are missing from the partnership`

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/LegalEntityDetailsSpec.scala [hunk 2]
```diff
@@ -66,6 +66,26 @@ class LegalEntityDetailsSpec
...
+        legalEntityDetails.customerDetails.individualDetails mustBe None
+
+        legalEntityDetails.customerDetails.organisationDetails.get.organisationType mustBe Some(
+          pptPartnershipDetails.organisationType.get.toString
+        )
+        legalEntityDetails.customerDetails.organisationDetails.get.organisationName mustBe "TODO"
+      }
     }
 
     "throw an exception" when {
```
Comment (tmc-ee) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/LegalEntityDetailsSpec.scala:91]: Can we add a test for exceptions? e.g. `incorporation details are missing from the partnership`

Also, I know this isn't part of the partnerships work per se, but do we also need to add test coverage for `organisation type is 'None'` here, i.e. replicating https://github.com/hmrc/plastic-packaging-tax-registration/blob/c746c2e2691ec878c45d8a6d7cd4571bdd6791b4/test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/CustomerDetailsSpec.scala#L68

**Review by tmc-ee — CHANGES_REQUESTED** [review]

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/CustomerDetailsSpec.scala (hunk 1 above)
Comment (RyanKellyHMRC) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/CustomerDetailsSpec.scala:66]: Additional test added.

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/LegalEntityDetailsSpec.scala (hunk 2 above)
Comment (RyanKellyHMRC) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/LegalEntityDetailsSpec.scala:91]: Additional tests added.

**Review by tmc-ee — APPROVED** [review]
Thanks for the changes, tests looking good! 👍

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/LegalEntityDetails.scala [hunk 3]
```diff
@@ -48,6 +48,14 @@ object LegalEntityDetails {
                              customerDetails = CustomerDetails(pptOrganisationDetails)
           )
         }.getOrElse(throw new Exception("Individual details are required"))
+      case Some(OrgType.PARTNERSHIP) =>
+        pptOrganisationDetails.partnershipDetails.map { details =>
+          LegalEntityDetails(dateOfApplication = getDateOfApplication,
+                             customerIdentification1 = details.sautr,
+                             customerIdentification2 = Some(details.postcode),
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscription/LegalEntityDetails.scala:55]: :+1:

**Review by PauloPortugal — APPROVED** [review]

**Review by kiranprakashdigital — APPROVED** [review]
